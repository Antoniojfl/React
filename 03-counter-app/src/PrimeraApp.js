import React from 'react';
import PropTypes from 'prop-types';

// Este import de React no es necesario en las nuevas versiones de react
//Los componentes se escriben con upper-cammel-case

//Functional components


const PrimeraApp = ({holaGoku,defaultProp}) =>{

    const saludo = 'Antonio';
    const saludo2 = 28;
    const saludo3 = true;
    const saludo4 = 12.26;
    const saludo5 = {
        name:'Antonio',
        age:28,
        gender:'Male'
    }

    //console.log(holaGoku);

    
    //Props

    /* Los props se pueden definir desde el render.dom pasandoselos al
    componente pero ya no se suele usar de esa manera, tambien
    se le puede hacer destructuring o se definen directamente en el componente */

    return (
    <>
        <h1>Hola Mundo !!!</h1>
        <h1>Imprimiendo un prop: {holaGoku}</h1>
        <h1>Puedo imprimir un string: {saludo}</h1>
        <h1>Puedo imprimir un number: {saludo2}</h1>
        <h1>NO puedo imprimir un boolean: {saludo3}</h1>
        <h1>Puedo imprimir un decimal: {saludo4}</h1>
        <pre>Puedo imprimir un objeto: {JSON.stringify(saludo5,null,3)}</pre>
        <p>Imprimiendo un defaultProp: {defaultProp}</p>
        <h1>Mi primera aplicacion</h1>

    </>  // o tambien se puede importar Fragment y escribir <Fragment> </Fragment>
    );
};

/* Para retornar mas de una linea de JSX
bien se puede retornar dentro de un div o un tag que los agrupe
segun necesidad, ejemplo si eso va a llevar alguna clase se
agrupa en un div y dentro del div se colocan los otros tags y se
le da estilo al div, pero si no se necesita hacer nada de eso se puede
encerrar en un fragment que es propio de React, se necesita importar,
tambien puede ir encerrado en parentesis para que react sepa que esta 
devolviendo un objeto y devuelve todo lo que esta en ese parentesis*/



//PROP-TYPES

/* esto es opcional pero nos permite definir las propiedades de nuestro
componente y opciones como si son requeridas o no, para que a futuro
quien trabaje con nuestro codigo se vea obligado a usarlas */

PrimeraApp.propTypes = {
    holaGoku: PropTypes.string.isRequired // Prop requerida
}

PrimeraApp.defaultProps = {
    defaultProp: 'Esta es una forma de poner default props'
}

export default PrimeraApp;