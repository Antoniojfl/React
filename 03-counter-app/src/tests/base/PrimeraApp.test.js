import React from 'react';
//import {getByText, render} from '@testing-library/react'
import {shallow} from 'enzyme'
import PrimeraApp from '../../PrimeraApp'


describe('Pruebas en <PrimeraApp />', () => {

   // test('debe de mostrar el mensaje "Hola Mundo" ', () => {
        
    // Forma 1
       /*  const  saludo = 'Hola Mundo';

        const wrapper = render(<PrimeraApp holaGoku='Hola soy Goku'/>);
        
        expect(wrapper.getByText(saludo)).toBeInTheDocument(); */

    /*Tambien se puede hacer de la siguente forma
    wrapper devuelve  un objeto el cual tiene la funcion getByText, se
    puede desestructurar y obtener la funcion directamente ejemplo: */

        /* const  saludo = 'Hola Mundo';

        const {getByText} = render(<PrimeraApp holaGoku='Hola soy Goku'/>);

        expect(getByText(saludo)).toBeInTheDocument(); */

    /*Se necesitÃ³ crear el archivo setupTests.js para que pudiera leer
    la dependencia donde esta el toBeInTheDocument(), el archivo lo lee
    el test suite cuando se corren las pruebas*/
 
   // })

   test('debe de mostrar el mensaje "Hola Mundo" con enzyme ', () => {

        const saludo = 'Hola soy Goku';
        const wrapper = shallow(<PrimeraApp holaGoku={saludo}/> );

        expect(wrapper).toMatchSnapshot();
        
   })

   test('debe de mostrar el sub ', () => {

    const saludo = 'Hola Mundo !!!';
    const propDos= 'Esta es una forma de poner default props';
    const wrapper = shallow(<PrimeraApp 
        holaGoku={saludo}
        defaultProp ={propDos}
        /> );

    const textoParrafo = wrapper.find('p').text();
    console.log(textoParrafo);

    expect(textoParrafo).toBe('Imprimiendo un defaultProp: Esta es una forma de poner default props');
    
})
   
    
    
})
 