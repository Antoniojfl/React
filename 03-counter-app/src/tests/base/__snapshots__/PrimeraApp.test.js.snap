// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe de mostrar el mensaje "Hola Mundo" con enzyme  1`] = `
<Fragment>
  <h1>
    Hola Mundo !!!
  </h1>
  <h1>
    Imprimiendo un prop: 
    Hola soy Goku
  </h1>
  <h1>
    Puedo imprimir un string: 
    Antonio
  </h1>
  <h1>
    Puedo imprimir un number: 
    28
  </h1>
  <h1>
    NO puedo imprimir un boolean: 
    <Component />
  </h1>
  <h1>
    Puedo imprimir un decimal: 
    12.26
  </h1>
  <pre>
    Puedo imprimir un objeto: 
    {
   "name": "Antonio",
   "age": 28,
   "gender": "Male"
}
  </pre>
  <p>
    Imprimiendo un defaultProp: 
    Esta es una forma de poner default props
  </p>
  <h1>
    Mi primera aplicacion
  </h1>
</Fragment>
`;
